### DISCUSSION TITLE
Can I download the source code, add my own custom rules, and compile to generate cli jar？

### DISCUSSION THREAD

**Discussion by opluscustom** [detekt/detekt#4109] [Q&A] [answered]
Can I generate cli all-jar Instead of using --plugin customRules.jar?
If I can, what should I do?

**Comment by BraisGabin** [comment, marked as answer]
Sure! clone the repo, do your changes and execute `./gradlew shadowJar`. That should work. (Remember to update the default.detekt-config.yml using `./gradlew generateDocumentation`)

But I'm curious. Why do you want to do that? What are you missing?

**Reply by opluscustom** [reply]
Thank you very much for your guidance！ Detekt functions very well ~
 Why do you want to do that? What are you missing?
--- Because  when use --plugin,we may need to introduce a lot jars，formating.jar、customRules1.jar/customRules1.jar etc.
